src/components/containers/ConnectedMellowLpDepositForm/ConnectedMellowLpDepositForm.tsx

```diff
@@ -0,0 +1,221 @@
+            setDepositState(DepositStates.APPROVE_FAILED);
+        })
+    };
+
+    useEffect(() => {
+        void vault.isTokenApproved().then((resp) => {
+            if (resp) {
+                setDepositState(DepositStates.APPROVED);
+            }
+            else {
+                setDepositState(DepositStates.APPROVE_REQUIRED);
+            }
+        }, (err: Error) => {
+            setError((<Text red> {`Error occured while retrieving information. ${err.message ?? ""}`}</Text>));
+            setDepositState(DepositStates.PROVIDER_ERROR);
+        });
+    }, [vault, selectedDeposit]);
+
+    let submissionState: {
+        submitText: string,
+        hintText: ReactNode,
+        action: () => void,
+        disabled: boolean,
+    } = {
+        submitText: "Initialising...",
+        action: () => { },
+        disabled: true,
+        hintText: "Initialising, please wait",
+    }
+
+    switch (depositState) {
+        case DepositStates.INITIALISING: {
+            submissionState = {
+                submitText: "Initialising...",
+                action: () => { },
+                hintText: "Initialising, please wait",
+                disabled: true,
+            }
+            break;
+        }
+        case DepositStates.PROVIDER_ERROR: {
+            submissionState = {
+                submitText: "Provider Error",
+                action: () => { },
+                hintText: error,
+                disabled: true,
+            }
+            break;
+        }
+        case DepositStates.APPROVE_FAILED: {
+            submissionState = {
+                submitText: "Approve",
+                action: approve,
+                hintText: error,
+                disabled: false,
+            }
+            break;
+        }
+        case DepositStates.DEPOSIT_FAILED: {
+            submissionState = {
+                submitText: "Deposit",
+                action: deposit,
+                hintText: error,
+                disabled: false,
+            }
+            break;
+        }
+        case DepositStates.APPROVE_REQUIRED: {
+            submissionState = {
+                submitText: "Approve",
+                action: approve,
+                hintText: `Please approve ${vault.tokenName}`,
+                disabled: false,
+            }
+            break;
+        }
+        case DepositStates.APPROVING: {
+            submissionState = {
+                submitText: "Approving...",
+                action: () => { },
+                hintText: "Waiting for confirmation",
+                disabled: true,
+            }
+            break;
+        }
+        case DepositStates.APPROVED: {
+            submissionState = {
+                submitText: "Deposit",
+                action: deposit,
+                hintText: (selectedDeposit > 0) ? (<><Text green>Tokens approved</Text>. Let's deposit!</>) : "Please input amount",
```

Comment:
is it not possible for hint to be just a string? I would not mix simple data structures with React.Node.

Reply:
Resolved. https://github.com/Voltz-Protocol/voltz-ui/pull/295/commits/a44c3c48e2a2660e1fc14f6eeeda646152d2edea